posts/2021-03-24-microprofile40-open-liberty-21003.adoc

```diff
@@ -0,0 +1,442 @@
+----
+
+To view the status of each health check, access the either the `http://<hostname>:<port>/health/live` or `http://<hostname>:<port>/health/ready endpoints`.
+
+More information:
+
+* link:https://github.com/eclipse/microprofile-health/releases/tag/3.0[MicroProfile Health 3.0 Release Page (Javadoc & Spec)]
+
+* link:https://download.eclipse.org/microprofile/microprofile-health-3.0/microprofile-health-spec-3.0.html#release_notes_3_0[Release notes]
+
+[#opentracing-2.0]
+=== New updates in  MicroProfile Open Tracing 2.0
+
+MicroProfile OpenTracing 2.0 can be used to profile and monitor applications built using microservice architecture.
+
+MicroProfile OpenTracing 2.0 has upgraded the OpenTracing API to version 0.33.0.  This allows the
+use of tracing backends and their libraries that are built on OpenTracing API 0.33.0.
+
+==== Try it now
+
+Include the following in the `server.xml`:
+
+[source, xml]
+----
+    <feature>mpOpenTracing-2.0</feature>
+----
+
+Also configure a tracing backend such as Jaeger or Zipkin. +
+For Jaeger, add the following maven dependencies in the application's pom.xml.
+
+[source, xml]
+----
+<dependency>
+    <groupId>io.jaegertracing</groupId>
+    <artifactId>jaeger-client</artifactId>
+    <version>1.2.0</version>
+</dependency>
+<dependency>
+    <groupId>org.slf4j</groupId>
+    <artifactId>slf4j-api</artifactId>
+    <version>1.7.30</version>
+</dependency>
+<dependency>
+    <groupId>org.slf4j</groupId>
+    <artifactId>slf4j-jdk14</artifactId>
+    <version>1.7.30</version>
+</dependency>
+----
+
+You can find out more about about configuring Jaeger settings using environment variables by looking
+at link:https://github.com/jaegertracing/jaeger-client-java/blob/v1.2.0/jaeger-core/README.md[jaeger-client-java readme].
+
+For the `JAEGER_PASSWORD` environment variable, the password can be encoded using the `securityUtility` command.
+
+Depending on Jaeger’s sampling settings `JAEGER_SAMPLER_TYPE` and `JAEGER_SAMPLER_PARAM`,
+Jaeger may not report every span created by the applications.
+
+For Zipkin, take a look at the link:https://github.com/WASdev/sample.opentracing.zipkintracer[sample project] to see how to implement a tracer for Liberty.
+
+Define your application in the `server.xml`:
+[source, xml]
+----
+<webApplication location="yourapp.war" contextRoot="/yourapp">
+    <!-- enable visibility to third party APIs -->
+    <classloader apiTypeVisibility="+third-party" />
+</webApplication>
+----
+
+Once you have hit some JAX-RS endpoints of your application, you should be able to find spans in the user interface of your tracing backend.
+
+More information:
+
+* link:https://github.com/eclipse/microprofile-opentracing/releases/tag/2.0[MicroProfile OpenTracing 2.0 Release Page (Javadoc & Spec)]
+
+* link:https://download.eclipse.org/microprofile/microprofile-opentracing-2.0/microprofile-opentracing-spec-2.0.html#_release_2_0[Release notes]
+
+[#open-api-2.0]
+=== What's New in MicroProfile Open API 2.0
```

Comment:
Should this be larger text? two rather than three equals signs?

Reply:
```suggestion
== What's New in MicroProfile Open API 2.0
```